storage/google_storage.py

```diff
@@ -3,53 +3,66 @@
 import json
 import os
 
-import google.cloud.storage
-import google.oauth2.service_account
+import google.cloud.storage  # type: ignore
+from google.cloud.storage.bucket import Bucket  # type: ignore
+from google.cloud.storage.blob import Blob  # type: ignore
+import google.oauth2.service_account  # type: ignore
 
-from . import retry
-from .storage import Storage, register_storage_protocol
+from typing import BinaryIO, cast, Optional
+
+from storage import retry
+from storage.storage import Storage, register_storage_protocol
+
+
+class GoogleStorageException(Exception):
+    pass
 
 
 @register_storage_protocol("gs")
 class GoogleStorage(Storage):
-    def _get_bucket(self):
-        credentials_data = json.loads(base64.urlsafe_b64decode(
-            self._parsed_storage_uri.username.encode("utf8")))
+
+    def _get_bucket(self) -> Bucket:
+        username = self._parsed_storage_uri.username
+        credentials_data = json.loads(base64.urlsafe_b64decode(username))
         credentials = google.oauth2.service_account.Credentials.from_service_account_info(
             credentials_data)
         client = google.cloud.storage.client.Client(
             project=credentials_data["project_id"], credentials=credentials)
         return client.get_bucket(self._parsed_storage_uri.hostname)
 
-    def _get_blob(self):
+    def _get_blob(self) -> Blob:
         bucket = self._get_bucket()
-        return bucket.blob(self._parsed_storage_uri.path[1:])
+        blob = bucket.blob(self._parsed_storage_uri.path[1:])
+        if blob is None:
```

Comment:
I don't think we're using it anymore, but just FYI `Bucket.get_blob()` will return None. This particular check for `None` isn't necessary, even for the type checker in this instance, since the result of any un-typed library / method / etc. is `Any`.